File: AlgorithmStudy/src/main/java/io/github/devwithpug/goorm_43064.java
// goorm Binary Search
// https://level.goorm.io/exam/43064/binary-search/quiz/1
package io.github.devwithpug;

import java.io.*;

public class goorm_43064 {

    private static void binary_search(int k, String[] arr, int a, int b) {
        if (a > b) {
            System.out.println("X");
            return;
        }

        int mid = (a + b) / 2;

        if (k < Integer.parseInt(arr[mid]))
            binary_search(k, arr, a, mid - 1);
        else if (k > Integer.parseInt(arr[mid]))
            binary_search(k, arr, mid + 1, b);
        else if (k == Integer.parseInt(arr[mid]))
            System.out.println(mid + 1);
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String[] arr = br.readLine().split(" ");
        int k = Integer.parseInt(br.readLine());

        binary_search(k, arr, 0, n - 1);
    }
}

File: python/goorm/탐색/goorm_43064.py
# goorm Binary search
# https://level.goorm.io/exam/43064/binary-search/quiz/1

# -*- coding: utf-8 -*-
# UTF-8 encoding when using korean

n = input()
arr = input().split(" ")
k = input()


def binary_search(k, arr, a, b):
    if a > b:
        print("X")
        return
    mid = int((a + b) / 2)
    if int(arr[mid]) < int(k):
        binary_search(k, arr, mid + 1, b)
    elif int(arr[mid]) > int(k):
        binary_search(k, arr, a, mid - 1)
    elif arr[mid] == k:
        print(mid + 1)


binary_search(k, arr, 0, n - 1)
